src/lib/y2storage/planned/partitions_distribution.rb

```diff
@@ -111,31 +117,73 @@ def partitions_count
       #
       # @return [Integer] -1, 0, 1 like <=>
       def better_than(other)
-        # The smallest gaps the better
-        res = gaps_total_disk_size <=> other.gaps_total_disk_size
-        return res unless res.zero?
+        criteria = [
+          # The smaller gaps the better
+          [:gaps_total_size, :smaller],
 
-        # The fewer gaps the better
-        res = gaps_count <=> other.gaps_count
-        return res unless res.zero?
+          # The fewer gaps the better
+          [:gaps_count, :smaller],
 
-        # The fewer physical volumes the better
-        res = partitions_count <=> other.partitions_count
-        return res unless res.zero?
+          # The fewer physical volumes the better
+          [:partitions_count, :smaller],
 
-        # The less fragmentation the better
-        res = spaces.count <=> other.spaces.count
-        return res unless res.zero?
+          # The bigger installation the better
+          [:spaces_total_size, :bigger],
 
-        # The biggest installation the better
-        res = other.spaces_total_disk_size <=> spaces_total_disk_size
-        return res unless res.zero?
+          # We want the partitions with biggest weight to be placed in the spaces
+          # in which there is more extra space to be distributed
+          [:weight_space_deviation, :smaller],
+
+          # The less fragmentation the better
+          [:spaces_count, :smaller]
```

Comment:
BTW, why not adding here also `[:comparable_string, :smaller]` ? so it is in one list.
and also it smell like a constant for me.

Reply:
Well, the previous loop contains a `return res unless res.zero?` as the last line. I wanted to empathize two things:

- That `unless res.zero?` does not affect `#comparable_string`. No matter if the result is zero (which in fact can NEVER happen for two distributions that are in fact different), the result will be returned.
- That `#comparable_string` is not a real criteria. It have no meaning about the value of the distribution for our purposes. It's just a last-resort fallback.